### DISCUSSION TITLE
Linux, Manager: problems found with alpha test version 8.0.4

### DISCUSSION THREAD

**Discussion by RichardHaselgrove** [BOINC/boinc#5787] [General]
**Describe the bug**
Items noticed when installing and running v8.0.4 on a new machine, with Linux Mint 22

I've dividing the list into two: the first part seems to comprise new bugs with this version, the second part may have been around for years but still niggle.

1. Whenever I open the Manager from terminal (command line) it reports:
    `Gtk-Message: 17:58:17.328: Failed to load module "xapp-gtk3-module"`
2. The file menu contains 5 entries: four (New window, Select computer, Close Window and Exit Manager) are active, but 'Shut down connected client ...' is disabled.
3. When I chose Exit Manager, this dialog pops up:
     ![image](https://github.com/user-attachments/assets/12cd220d-a646-43f0-a54f-abcf379125bc)
4. I have enabled remote monitoring, and I can shut down the connected client from remote machines - which seems inconsistent.
5. I have been struggling all day to establish a complete set of configuration files, with the primary client files, on a separate data drive. I've got that now, and it seems (mostly) consistent, but it felt like playing whack-a-mole - every time I fixed one thing, another broke.
6. One particularly difficult one to track down was that 'something' triggered it to create a minimal set of configuration files in my home directory. They were missing the GUI RPC password I'd set up, so the manger couldn't connect to the client until I worked out what had happened, and deleted the extra files. Since then, it's been OK, and is processing tasks as I type.

Part 2 - things which have probably been around for years:

1. When using the 'Add project ...' tool, the height of the email entry box is too low - letters with descenders (g, q, y) are truncated and hard to proof-read.
2. The fonts used in the 'Computing preferences' dialog (bold italic) are too large and blocky: some numbers (like 100.00%) overflow the entry boxes, and some labels are truncated. This doesn't affect Windows managers, but has been present in Linux for as long as I've been running it: it's become more apparent since the division into 'in use' and 'not in use' sections.

I'll keep watching for other anomalies, but that will do for now.

**System Information**
 - OS: Linux Mint 22
 - BOINC Version: 8.0.4 alpha

**Comment by AenBleidd** [comment]
Part 1

1. This could be ignored. We are building BOINC statically to avoid dependency hell.
2. This is intentional: be default, we are running BOINC client on linux using `systemctl`. To run/restart/shutdown applications running is such a way you need to have `sudo` permissions. When BOINC Manager is running under regular user's account, we can shutdown the client, that's true, because we can just send the command to the client to shutdown itself, but then there is no way to start BOINC client again without restarting the PC. That is why this functionality was disabled.
3. The same as Point 2.
4. True, but this is a necessary evil: users who do this way are usually more experienced, and they need to be aware about the consequences of such actions (btw, good point, will add this to the documentation).
5. Without any details I can't say what exactly is misleading.
6. Again, need more details, and probably the same as Point 5, please create a separate ticket to discuss and fix particular issues, that's much easier.

Part 2

1. Please report a separate ticket. Will try to fix that.
2. Yes, that's true. Please create a separate ticket as well and maybe once I will have time to dive into this again to fix that.

Thank you in advance.

P.S. Once the separate tickets will be created - I'll close this one.

**Comment by RichardHaselgrove** [comment]
Separate tickets for Part 2 created (#5785, #5786).

I'll convert this ticket to a discussion - please don't close it just yet, as I think we still need to talk about Part 1.

**Comment by RichardHaselgrove** [comment]
Comments on Part 1:

1. I agree - it can be ignored, but it should still be tidied up. It doesn't inspire confidence that this is a professional product.
2. OK, I've investigated this more thoroughly. I've been using systemd as long as I've been using Linux (about 5 years), so I'm familiar with the management commands. I have been using:
sudo systemctl stop boinc-client
sudo systemctl disable boinc-client
sudo systemctl enable boinc-client
sudo systemctl start boinc-client
before and after making major changes. The new installer has renamed it "boinc-client.service", but I can adapt to that.
3. And I find that if the service is disabled, and you open BOINC Manager, **it starts the client**! That's actually quite neat, but it should be documented. When the client is started that way, the exit dialog changes to:
![Screenshot at 2024-08-31 10-09-59](https://github.com/user-attachments/assets/9c0d96f2-55f6-4053-802c-9f5cadf3addd)
That makes more sense, but the original, default, setting doesn't.
4. OK, we can leave that as an oddity for experienced users to play with.
5. I'll watch for this as the machine settles down, and try to pin it down if it recurs.
6. ditto.

There needs to be a very thorough review of the user manual in the Wiki after this initiative. In particular.
 https://github.com/BOINC/boinc/wiki/Installing_on_Linux
(Fedora and Debian/Ubuntu have a very brief reference to the "new recommended way": Gentoo is apparently unchanged, and openSUSE takes you straight to the download instructions with no details about what it actually does)

https://github.com/BOINC/boinc/wiki/Stop_or_start_BOINC_daemon_after_boot
Since the new version seems to be dual-use, as discussed above, this would seem to be a good place to explain the pros and cons, and give details of how to use them. The current page seems hopelessly outdated - there's no mention of systemctl anywhere.
